### PR TITLE
CI and linting with pylint and flake8

### PR DESCRIPTION
**Pull request by nfnty**
You also need to register the repository at https://travis-ci.org

Most of the ignored flake8 error codes have to do with you not being PEP8 compliant in using 2 spaces indent instead of 4.

### CODE REVIEW

File: rplugin/python/lldb_nvim/controller.py [hunk 1]
```diff
@@ -1,14 +1,45 @@
...
+
+def complete_command(arg, line, pos):  # pylint: disable=too-many-branches
+  """ Returns a list of viable completions for line, and cursor at pos. """
+  pos = int(pos)
+  result = lldb.SBStringList()
+
+  if arg == line and line != '':
+    # provide all possible completions when completing 't', 'b', 'di' etc.
+    cands = ['']
+    for x in result:
```
Comment (nfnty) [inline rplugin/python/lldb_nvim/controller.py]: The only difference to `complete_command` is that it was moved out of the class due to having no `self` use. Nothing has changed from previous behavior.

File: rplugin/python/lldb_nvim/controller.py [hunk 2]
```diff
@@ -98,32 +132,6 @@ def safe_exit(self):
...
     self.join()
 
-  def complete_command(self, arg, line, pos):
-    """ Returns a list of viable completions for line, and cursor at pos. """
-    pos = int(pos)
-    result = lldb.SBStringList()
-
-    if arg == line and line != '':
-      # provide all possible completions when completing 't', 'b', 'di' etc.
-      num = self._ipreter.HandleCompletion('', 0, 1, -1, result)
```
Comment (critiqjo) [inline rplugin/python/lldb_nvim/controller.py]: > The only difference to `complete_command` is that it was moved out of the class due to having no `self` use. Nothing has changed from previous behavior.

`self` was used here to invoke `HandleCompletion` which populates `result`. I know it is a very non-pythonic API and might be confusing... `num` is unused, but that's not what it is used for here...

Comment (nfnty) [inline rplugin/python/lldb_nvim/controller.py]: Ah, now I understand, completely missed that. I removed those lines without thinking when pylint complained about unused variables. Which made it complain about no `self` use. The return can just be thrown away. I'll fix it.

**Comment by nfnty** [comment]
Travis will actually read the `.travis.yml` from the PR itself, you can see that the checks have passed.

File: rplugin/python/lldb_nvim/check_lldb.py [hunk 3]
```diff
@@ -1,20 +1,25 @@
...
-import os, sys
+from __future__ import (absolute_import, division, print_function)
+
+import os
+import sys
+
+__metaclass__ = type  # pylint: disable=invalid-name
 
-def probe():
+def probe():  # pylint: disable=too-many-branches
```
Comment (critiqjo) [inline rplugin/python/lldb_nvim/check_lldb.py]: Why not disable `too-many-branches` globally? I don't think there is an easy way to fix a majority of them. Or is there a parameter that can be tweaked to control the strictness?

File: rplugin/python/lldb_nvim/__init__.py [hunk 4]
```diff
@@ -1,23 +1,29 @@
+from __future__ import (absolute_import, division, print_function)
+
 import logging
 import neovim
-import check_lldb
+from . import check_lldb
+
+__metaclass__ = type  # pylint: disable=invalid-name
```
Comment (nfnty) [inline rplugin/python/lldb_nvim/__init__.py]: It's basically a way of making all classes in Python 2 "new style classes", like the default behavior of Python 3.

File: rplugin/python/lldb_nvim/controller.py [hunk 5]
```diff
@@ -1,14 +1,20 @@
...
 from .vim_buffers import VimBuffers
 from .session import Session
 
+__metaclass__ = type  # pylint: disable=invalid-name
+
 class EventLoopError(Exception):
   pass
 
-class Controller(Thread):
+
```
Comment (nfnty) [inline rplugin/python/lldb_nvim/controller.py]: Leftover from before I turned off the error in flake8. flake8 normally complains if there's less than 2 newlines between definitions.

File: rplugin/python/lldb_nvim/check_lldb.py (hunk 3 above)
Comment (nfnty) [inline rplugin/python/lldb_nvim/check_lldb.py]: Potentially could increase `max-branches` (12 by default), but ideally this should be refactored in the future.

File: rplugin/python/lldb_nvim/__init__.py (hunk 4 above)
Comment (nfnty) [inline rplugin/python/lldb_nvim/__init__.py]: I.e. no need to define classes like `class MyClass(object):` anymore for new style classes, `class Myclass:` is enough.

**Comment by critiqjo** [comment]
:+1: Alright! Looks good!

**Comment by nfnty** [comment]
:ok_hand:

What do you think about converting to using 4 space indent to be fully compliant with PEP8?

**Comment by critiqjo** [comment]
> What do you think about converting to using 4 space indent to be fully compliant with PEP8?

Sure! :+1:

**Pull request merged**